### PR TITLE
Feature/activity class

### PR DESCRIPTION
**Pull request by itsnameissteven**
### Description
* Is this a feature or a fix?
This is a fix
* Where should the reviewer start?
Review the activity test and activity.js file
* What functionalities do these changes effect? Why were these changes made? What was the motivation behind these changes?
Activity Test was refactored utilizing less data.  Added additional tests 
* What outside features or research did you use to implement this fix?
n/a
* How should this be tested?
Run npm test to make sure all tests still pass. Open web app to ensure everything still loads properly
### Relevant Screen Shots
n/a
***
#### Please review considerations below:
- [X] Follows Turing Style Guidelines
- [X] Changes generate no new warnings or errors
- [X] Checked code for and corrected any spelling errors
- [ ] README has been updated(if applicable)

### CODE REVIEW

File: src/Activity.js [hunk 1]
```diff
@@ -65,23 +76,25 @@ class Activity {
...
+    return this.showChallengeListAndWinner(user, date, userRepo)[0];
   }
+
   getStreak(userRepo, id, relevantData) {
     let data = this.activityData;
     let sortedUserArray = (userRepo.makeSortedUserArray(id, data)).reverse();
-    let streaks = sortedUserArray.filter(function(element, index) {
-      if (index >= 2) {
-        return (sortedUserArray[index - 2][relevantData] < sortedUserArray[index - 1][relevantData] && sortedUserArray[index - 1][relevantData] < sortedUserArray[index][relevantData])
+    let streaks = sortedUserArray.reduce((acc, userData, index, arr) => {
```
Comment (aemiers) [inline src/Activity.js]: Can these be consts or is let the best way to go?

File: src/Activity.js [hunk 2]
```diff
@@ -65,23 +76,25 @@ class Activity {
...
-    return streaks.map(function(streak) {
-      return streak.date;
-    })
+       return acc;
+    },[]);
+    return streaks;
   }
+
   getWinnerId(user, date, userRepo) {
     let rankedList = this.getFriendsAverageStepsForWeek(user, date, userRepo);
```
Comment (aemiers) [inline src/Activity.js]: Const?

File: src/Activity.js (hunk 1 above)
Comment (aemiers) [reply src/Activity.js]: NOICE REDUCE!

File: src/Activity.js [hunk 3]
```diff
@@ -2,40 +2,49 @@ class Activity {
   constructor(activityData) {
     this.activityData = activityData
   }
+
   getMilesFromStepsByDate(id, date, userRepo) {
     let userStepsByDate = this.activityData.find(data => id === data.userID && date === data.date);
```
Comment (alexthompson207) [inline src/Activity.js]: maybe this could be a `const`?

**Review by alexthompson207 — COMMENTED** [review]
Good work with the function refactoring!

File: src/Activity.js (hunk 1 above)
Comment (itsnameissteven) [reply src/Activity.js]: done

**Pull request merged**